signing/signer.go

```diff
@@ -22,7 +22,11 @@ func NewPublicKey(pub []byte) *PublicKey {
 
 // Bytes returns the public key as byte array
 func (p *PublicKey) Bytes() []byte {
-	return p.pub
+	// Prevent segfault if unset
+	if p != nil {
+		return p.pub
+	}
+	return []byte{}
```

Comment:
As noted in a previous comment, a `nil` go slice is equivalent to an empty slice. This change should be undone.

Reply:
Please look at this again. The problem here isn't if `p.pub` is nil. The problem is if `p` itself is nil, which causes a segfault. I changed the last line here to return `nil` instead.

Reply:
Hmm, I see... I really misread that. I guess it's fine now, then.